feature/matcher/src/test/java/com/simprints/matcher/usecases/CreateRangesUseCaseTest.kt

```diff
@@ -10,42 +10,259 @@ class CreateRangesUseCaseTest {
-        assertThat(useCase.invoke(1, 5)).isEqualTo(listOf(0..1))
+    fun `should handle totalCount equal to MAX_BATCH_SIZE`() {
+        // When
+        val result = useCase(2000, 1)
+
+        // Then
+        assertThat(result).containsExactly(0 until 2000).inOrder()
     }
 
     @Test
-    fun `Returns single item if max withing single batch`() {
-        assertThat(useCase.invoke(20, 25)).isEqualTo(listOf(0..20))
+    fun `should handle batch sizes that are exactly MAX_BATCH_SIZE`() {
+        // When
+        val result = useCase(4000, 2)
+
+        // Then
+        assertThat(result).containsExactly(
+            0 until 2000,
+            2000 until 4000
+        ).inOrder()
     }
 
     @Test
-    fun `Correctly calculates last batch reminder`() {
-        assertThat(useCase.invoke(17, 10)).isEqualTo(
-            listOf(
-                0..10,
-                10..17,
-            ),
-        )
+    fun `should create correct ranges when numCandidates is less than numConsumers`() {
+        // When
+        val result = useCase(3, 5)
+
+        // Then
+        assertThat(result).containsExactly(
+            0..0,
+            1..1,
+            2..2
+        ).inOrder()
     }
 
     @Test
-    fun `Correctly calculates ranges for exact batches`() {
-        assertThat(useCase.invoke(210, 10)).isEqualTo(
-            listOf(
-                0..10, // size=10
-                10..20, // size=10
-                20..40, // size=20
-                40..70, // size=30
-                70..110, // size=40
-                110..160, // size=50
-                160..210, // size=50
-            ),
-        )
+    fun `should create correct ranges with uneven distribution`() {
+        // When
+        val result = useCase(11, 3)
+
+        // Then
+        assertThat(result).containsExactly(
+            0..3,
+            4..7,
+            8..10
+        ).inOrder()
+    }
+
+    @Test
+    fun `should create empty list when numCandidates is zero`() {
+        // When
+        val result = useCase(0, 5)
+
+        // Then
+        assertThat(result).isEmpty()
+    }
+
+    @Test
+    fun `should create single range when numConsumers is one`() {
+        // When
+        val result = useCase(10, 1)
+
+        // Then
+        assertThat(result).containsExactly(0..9)
+    }
+
+    @Test
+    fun `should handle large numbers correctly`() {
+        // When
+        val result = useCase(1000, 4)
+
+        // Then
+        assertThat(result).containsExactly(
+            0..249,
+            250..499,
+            500..749,
+            750..999
+        ).inOrder()
+    }
+
+    @Test
+    fun `should handle 2500 candidates with 4 processors`() {
+        // When
+        val result = useCase(2500, 4)
+
+        // Then
+        // 4 processors, batches under 2000 each, so 4 total batches
+        // Base size = 2500/4 = 625, remainder = 0
+        assertThat(result).containsExactly(
+            0 until 625,
```

Comment:
Test cases before this one use the `rangeTo` operator; it would be great to have it consistent, does not matter which way.